src/app/product-list/product-info/product-info.component.ts

```diff
@@ -13,16 +16,12 @@ import { Observable } from 'rxjs';
 export class ProductInfoComponent implements OnInit {
   product$: Observable<Product>;
 
-  constructor(
-    private productService: ProductService,
-    private route: ActivatedRoute
-  ) {}
+  constructor(private route: ActivatedRoute, private store: Store<AppState>) {}
 
   ngOnInit() {
-    this.product$ = this.route.params.pipe(
-      switchMap((parameter: Params) =>
-        this.productService.getProduct$(parameter.id)
-      )
-    );
+    this.route.params.pipe(take(1)).subscribe((parameter: Params) => {
```

Comment:
you want to update selectedProduct EACH TIME when the url changes... 
remove take(1) and move out this.product$ from the subscribe